cp_test_nn/predict.py

```diff
@@ -1,23 +1,39 @@
+import sys
+
 from cp_test_nn import process
 from cp_test_nn.dataset import create_dataset
 
-def test_cv_success_rate(nn, tokens, cv_file):
+def predict(nn, tokens, cv_file):
     data = create_dataset(process.failures(cv_file), tokens)
     X = data['dataset']
     y = data['results']
 
+    # Statistics that get triggered when we have result data
     successes = 0
     false_negatives = 0
     false_positives = 0
-    total_count = len(X)
+    total_count = 0
+
     for i, item in enumerate(X):
         pred = nn.predict([item])
-        if pred == [y[i]]:
-            successes += 1
-        elif pred == [1]:
-            false_positives += 1
-        else:
-            false_negatives += 1
-    print('Success rate: ', successes/total_count)
-    print('False positives: ', false_positives/total_count)
-    print('False negatives: ', false_negatives/total_count)
+        sys.stdout.write("{0} {1} {2} {3} {4}\n".format(
+            pred and "FLAKE  " or "UNKNOWN",
```

Comment:
So `pred` is a list of predictions - since this is a prediction for one item, it will be a list containing exactly one prediction. It will always be 0 or 1, since we're only predicting membership of two classes - "flake" (`pred == [1]`) and "not flake" (`pred == [0]`).
I think what we really want here now is to do `pred = nn.predict([item])[0]` on the line above and then you should be able to do this (+ also replace "UNKNOWN" for "NOT FLAKE" or something similar).

Does this make sense?